### PR TITLE
Added --without-error flag

### PR DESCRIPTION
**Pull request by daubaris**
Regarding #26. I've added the `--without-error` flag. It might need another name though. 

In addition, I've split some functionality from the `main` function so that it would be possible to test the added feature.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -34,6 +34,9 @@ struct Options {
     #[structopt(short = "t", long = "cargo-toml", default_value = "./Cargo.toml")]
     /// Pass the path of the `Cargo.toml` file
     cargo_toml: String,
+    #[structopt(short = "W", long = "without-error")]
```
Comment (o0Ignition0o) [inline src/main.rs]: could we use a lowercased short handle ? "w" seems fine to me :)

File: src/main.rs [hunk 2]
```diff
@@ -67,15 +70,46 @@ fn main() -> Result<(), error::Error> {
...
+    #[test]
+    fn test_return_status_with_lints() {
+        let lints = vec![Lint {
+            package_id: "cargo-scout".to_string(),
+            src_path: None,
+            message: None,
+        }];
+
+        assert!(return_warnings(&lints, true).is_ok());
+        assert!(return_warnings(&lints, false).is_err());
```
Comment (o0Ignition0o) [inline src/main.rs:104]: You thought of adding tests, that's awesome ! 🎉

File: src/main.rs [hunk 3]
```diff
@@ -67,15 +70,46 @@ fn main() -> Result<(), error::Error> {
 
     let warnings_caused_by_diff =
         intersections::get_lints_from_diff(&clippy_lints, &diff_sections, opts.verbose);
-    if warnings_caused_by_diff.is_empty() {
+
+    return_warnings(&warnings_caused_by_diff, opts.without_error)
```
Comment (o0Ignition0o) [inline src/main.rs:75]: Refactoring it into an other function is a great idea !

File: src/main.rs [hunk 4]
```diff
@@ -67,15 +70,46 @@ fn main() -> Result<(), error::Error> {
...
         println!("No warnings raised by clippy::pedantic in your diff, you're good to go!");
         Ok(())
     } else {
-        display_warnings(&warnings_caused_by_diff);
-        println!(
-            "Clippy::pedantic found {} warnings",
-            warnings_caused_by_diff.len()
-        );
+        if without_error {
+            display_warnings(&lints);
```
Comment (o0Ignition0o) [inline src/main.rs]: It seems we call display_warning(&lints) whether without_error is true or false, maybe we can call it just once, before the if statement ?

File: src/main.rs [hunk 5]
```diff
@@ -67,15 +70,46 @@ fn main() -> Result<(), error::Error> {
...
+            return Ok(());
+        }
+        display_warnings(&lints);
+        println!("Clippy::pedantic found {} warnings", lints.len());
```
Comment (o0Ignition0o) [inline src/main.rs]: I like displaying the warnings either way, could we maybe move it to the display_warnings function ?

**Review by o0Ignition0o — APPROVED** [review]
It looks great !
You even thought of adding tests, which is really awesome 🎉 
A couple of nits, and it can get merged ! :)

File: src/main.rs (hunk 4 above)
Comment (daubaris) [reply src/main.rs]: That's true, somehow missed that..

**Comment by o0Ignition0o** [comment]
That's perfect ! Merging it right now, thanks for your great work ! 🎉

**Pull request merged**